butterknife-sample-lib/build.gradle

```diff
@@ -0,0 +1,31 @@
+buildscript {
+  repositories {
+    mavenLocal()
+    mavenCentral()
+  }
+  dependencies {
+    classpath "${GROUP}:butterknife-gradle-plugin:${VERSION_NAME}"
+  }
+}
+
+apply plugin: 'com.android.library'
+apply plugin: 'com.neenbedankt.android-apt'
+apply plugin: 'butterknife'
```

Comment:
Libraries depend on the plugin so no issue there. For apps that don't depend on it; "if you want nicer generated code, add the `@RClass` annotation" and "if you want nicer generated code, add the plugin" sounds like equivalent effort for the consumer.

Moving the responsibility to the plugin does mean it needs to be replicated for each build system (Buck, Bazel etc.). The `variant.javaCompile.options` method might not be as straightforward, but "generate a file with information" should be simple enough to do for each?

Native support by android-tools would definitely be awesome. Parsing the manifest and deriving information always feels a bit hacky.